### PR TITLE
Slider does not show changed label value for keyboard users fix

### LINKED ISSUE #152884: Slider does not show changed label value for keyboard users
### Steps to reproduce

1. Create a simple Slider widget (using the Slider sample code from documentation: https://api.flutter.dev/flutter/material/Slider-class.html)
3. Focus on the slider by tabbing to it
4. Click up, down, left, or right arrow key to change current value
5. Notice that the value indicator label is not shown when value changes

### Expected results

The value indicator label should be shown to the user using arrow keys when its value is changed just like it is shown for drag and click actions.

This creates an accessibility issue for users that don't use a screen reader or mouse and only use the keyboard to traverse the website. Since drag and click actions can only be accomplished by a mouse, keyboard users do not have a way of knowing what the current value of the slider is when it's changed using arrow keys.

### Actual results

Value indicator label is not shown for users that change the current value using arrow keys.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

/// Flutter code sample for [Slider].

void main() => runApp(const SliderApp());

class SliderApp extends StatelessWidget {
  const SliderApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      home: SliderExample(),
    );
  }
}

class SliderExample extends StatefulWidget {
  const SliderExample({super.key});

  @override
  State<SliderExample> createState() => _SliderExampleState();
}

class _SliderExampleState extends State<SliderExample> {
  double _currentSliderValue = 20;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: const Text('Slider')),
      body: Slider(
        value: _currentSliderValue,
        max: 100,
        divisions: 5,
        label: _currentSliderValue.round().toString(),
        onChanged: (double value) {
          setState(() {
            _currentSliderValue = value;
          });
        },
      ),
    );
  }
}
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Screen recording 2024-08-05 12.16.24 PM.webm](https://github.com/user-attachments/assets/806e94d7-7c6c-40e5-93af-dbf2fb9e0dd4)

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[!] Flutter (Channel [user-branch], 3.24.0-1.0.pre.280, on Debian GNU/Linux rodete 6.6.15-2rodete2-amd64, locale
    en_US.UTF-8)
    ! Flutter version 3.24.0-1.0.pre.280 on channel [user-branch] at /usr/local/google/home/denisbowen/flutter
      Currently on an unknown channel. Run `flutter channel` to switch to an official channel.
      If that doesn't fix the issue, reinstall Flutter by following instructions at https://flutter.dev/setup.
    ! Upstream repository https://github.com/DBowen33/flutter.git is not a standard remote.
      Set environment variable "FLUTTER_GIT_URL" to https://github.com/DBowen33/flutter.git to dismiss this error.
    • Framework revision 993593818f (13 days ago), 2024-07-23 16:43:20 +0000
    • Engine revision 74737820a8
    • Dart version 3.6.0 (build 3.6.0-78.0.dev)
    • DevTools version 2.37.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to
      perform update checks and upgrades.

[!] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
    • Android SDK at /usr/local/google/home/denisbowen/Android/Sdk
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/to/linux-android-setup for more details.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✗] Linux toolchain - develop for Linux desktop
    • Debian clang version 16.0.6 (26)
    ✗ CMake is required for Linux development.
      It is likely available from your distribution (e.g.: apt install cmake), or can be downloaded from
      https://cmake.org/download/
    ✗ ninja is required for Linux development.
      It is likely available from your distribution (e.g.: apt install ninja-build), or can be downloaded from
      https://github.com/ninja-build/ninja/releases
    ✗ pkg-config is required for Linux development.
      It is likely available from your distribution (e.g.: apt install pkg-config), or can be downloaded from
      https://www.freedesktop.org/wiki/Software/pkg-config/

[✓] Android Studio (version 2023.2)
    • Android Studio at /opt/android-studio-with-blaze-2023.2
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.9+0-17.0.9b1087.7-11185874)

[✓] VS Code (version 1.87.1)
    • VS Code at /usr/share/code
    • Flutter extension version 3.92.0

[✓] Connected device (2 available)
    • Linux (desktop) • linux  • linux-x64      • Debian GNU/Linux rodete 6.6.15-2rodete2-amd64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 127.0.6533.88

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 3 categories.
```

</details>

### PR DESCRIPTION
**Pull request by DBowen33**
Fix issue to where keyboard users could not see visual indicator label of changed value in Slider

Before: 
[Screen recording 2024-08-05 12.16.24 PM.webm](https://github.com/user-attachments/assets/89b99423-7856-4b25-86de-b211b2dbe178)

After:
[Screen recording 2024-08-05 12.38.20 PM.webm](https://github.com/user-attachments/assets/641f9065-8279-4b79-89b1-b5bcd3d691a8)

Fixes https://github.com/flutter/flutter/issues/152884
Fixes b/340638215

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

File: packages/flutter/lib/src/material/slider.dart [hunk 1]
```diff
@@ -1818,6 +1818,7 @@ class _RenderSlider extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
       final double increase = increaseValue();
       onChanged!(increase);
       onChangeEnd!(increase);
+      _state.showValueIndicator();
```
Comment (QuncCccccc) [inline packages/flutter/lib/src/material/slider.dart:1821]: Should we check whether _state is mounted before? Like:

```dart
    if (!_state.mounted) {
      return;
    }
```

**Review by QuncCccccc — COMMENTED** [review]
Thanks for the fix! Looks good to me, just left one question.

File: packages/flutter/lib/src/material/slider.dart (hunk 1 above)
Comment (DBowen33) [reply packages/flutter/lib/src/material/slider.dart:1824]: Agreed! Added check like code snippet in lines 1493 - 1495.

**Review by QuncCccccc — APPROVED** [review]
LGTM:) Thanks!

**Review by Renzo-Olivares — APPROVED** [review]
LGTM, thanks for the fix!

**Review by MitchellGoodwin — COMMENTED** [review]
Code looks good to me. But should we wait to show the label until they use arrow keys, or should we show it right away once the slider is focused? If somebody using keyboard interaction just wanted to see the label, they'd have to go either up or down first which feels awkward. Also, in the video the label appearing briefly before moving to the next value looks jerky. Showing the label on focus seems like it would be smoother, but not doing so originally seemed intentional.

**Comment by QuncCccccc** [comment]
> should we wait to show the label until they use arrow keys, or should we show it right away once the slider is focused?

Thanks for catching this! I just checked the Material Design website, seems based on the [specs](https://m3.material.io/components/sliders/accessibility#f614609c-43c9-468c-a517-4d17628201d8) here, we probably should show the label right away once the slider is focused.

**Comment by DBowen33** [comment]
> > should we wait to show the label until they use arrow keys, or should we show it right away once the slider is focused?
> 
> Thanks for catching this! I just checked the Material Design website, seems based on the [specs](https://m3.material.io/components/sliders/accessibility#f614609c-43c9-468c-a517-4d17628201d8) here, we probably should show the label right away once the slider is focused.

Thanks for the catch as well! Created a new issue (https://github.com/flutter/flutter/issues/152961) with corresponding PR (https://github.com/flutter/flutter/pull/152960) to show label to user on focus

**Pull request merged**